script/bin/operator.rs

```diff
@@ -281,33 +569,93 @@ fn get_block_update_interval() -> u64 {
         .unwrap_or(360)
 }
 
+/////// Chain Config ///////
+
+#[derive(Debug, serde::Deserialize)]
+struct ChainConfig {
+    rpc_url: String,
+    blobstream_address: Address,
+}
+
+impl ChainConfig {
+    /// Tries to read from the `CHAINS_PATH` environment variable, then the default path (`../chains.json`).
+    ///
+    /// If neither are set, it will try to use [`Self::from_env`].
+    fn fetch() -> Result<Vec<Self>> {
+        const DEFAULT_PATH: &str = "chains.json";
+
+        let path = env::var("CHAINS_PATH").unwrap_or(DEFAULT_PATH.to_string());
+
+        Self::from_file(&path).or_else(|_| {
+            tracing::info!("No chains file found, trying env.");
+            Self::from_env().map(|c| vec![c])
+        })
+    }
+
+    /// Tries to read from the `CONTRACT_ADDRESS` and `RPC_URL` environment variables.
+    fn from_env() -> Result<Self> {
+        let address = env::var("CONTRACT_ADDRESS").context("CONTRACT_ADDRESS not set")?;
+        let rpc_url = env::var("RPC_URL").context("RPC_URL not set")?;
+
+        Ok(Self {
+            rpc_url,
+            blobstream_address: address.parse()?,
+        })
+    }
+    fn from_file(path: &str) -> Result<Vec<Self>> {
+        tracing::debug!("Reading chains from file: {}", path);
+
+        let file = std::fs::read_to_string(path)?;
+
+        Ok(serde_json::from_str(&file)?)
+    }
+}
+
 #[tokio::main]
 async fn main() {
     dotenv::dotenv().ok();
-    env_logger::init();
 
-    let operator = SP1BlobstreamOperator::new().await;
+    // Setup tracing.
+    tracing_subscriber::fmt::fmt()
+        .with_env_filter(
+            EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::from_env("info")),
+        )
+        .init();
+
+    // Succinct deployments use the `CHAINS` environment variable.
+    let config = ChainConfig::fetch().expect("Failed to fetch chain config");
+    let maybe_private_key: Option<PrivateKeySigner> = env::var("PRIVATE_KEY")
+        .ok()
+        .map(|s| s.parse().expect("Failed to parse PRIVATE_KEY"));
+
+    // Setup the KMS relayer config.
+    let signer_mode = env::var("USE_KMS_RELAYER")
+        .map(|s| s.parse().expect("USE_KMS_RELAYER failed to parse"))
+        .unwrap_or(SignerMode::Kms);
+
+    // Ensure a signer is set if KMS relayer is false.
+    if matches!(signer_mode, SignerMode::Local) && maybe_private_key.is_none() {
+        panic!("PRIVATE_KEY is not set but signer mode is local.");
+    }
```

Comment:
Nice! We can close this PR now: https://github.com/succinctlabs/sp1-blobstream/pull/22